qiskit_nature/circuit/library/ansatzes/ucc.py

```diff
@@ -0,0 +1,356 @@
+            alpha_spin: boolean flag whether to include alpha-spin excitations.
+            beta_spin: boolean flag whether to include beta-spin excitations.
+            max_spin_excitation: the largest number of excitations within a spin. E.g. you can set
+                                 this to 1 and `num_excitations` to 2 in order to obtain only
+                                 mixed-spin double excitations (alpha,beta) but no pure-spin double
+                                 excitations (alpha,alpha or beta,beta).
+            reps: The number of times to repeat the evolved operators.
+            initial_state: A `QuantumCircuit` object to prepend to the circuit.
+        """
+        self._qubit_converter = qubit_converter
+        self._num_particles = num_particles
+        self._num_spin_orbitals = num_spin_orbitals
+        self._excitations = excitations
+        self._alpha_spin = alpha_spin
+        self._beta_spin = beta_spin
+        self._max_spin_excitation = max_spin_excitation
+
+        super().__init__(reps=reps, evolution=PauliTrotterEvolution(), initial_state=initial_state)
+
+        # We cache these, because the generation may be quite expensive (depending on the generator)
+        # and the user may want quick access to inspect these. Also, it speeds up testing for the
+        # same reason!
+        self._excitation_ops: List[SecondQuantizedOp] = None
+
+    @property
+    def qubit_converter(self) -> QubitConverter:
+        """The qubit operator converter."""
+        return self._qubit_converter
+
+    @qubit_converter.setter
+    def qubit_converter(self, conv: QubitConverter) -> None:
+        """Sets the qubit operator converter."""
+        self._invalidate()
+        self._qubit_converter = conv
+
+    @property
+    def num_spin_orbitals(self) -> int:
+        """The number of spin orbitals."""
+        return self._num_spin_orbitals
+
+    @num_spin_orbitals.setter
+    def num_spin_orbitals(self, n: int) -> None:
+        """Sets the number of spin orbitals."""
+        self._invalidate()
+        self._num_spin_orbitals = n
+
+    @property
+    def num_particles(self) -> Tuple[int, int]:
+        """The number of particles."""
+        return self._num_particles
+
+    @num_particles.setter
+    def num_particles(self, n: Tuple[int, int]) -> None:
+        """Sets the number of particles."""
+        self._invalidate()
+        self._num_particles = n
+
+    @property
+    def excitations(self) -> Union[str, int, List[int], Callable]:
+        """The excitations."""
+        return self._excitations
+
+    @excitations.setter
+    def excitations(self, exc: Union[str, int, List[int], Callable]) -> None:
+        """Sets the excitations."""
+        self._invalidate()
+        self._excitations = exc
+
+    # TODO: the `preferred_init_points`-implementation can (and should!) be improved!
+    @property
+    def preferred_init_points(self):
```

Comment:
Do we want this to be on the `EvolvedOpAnsatz` level now that we moved `initial_state` down there, too?

Reply:
I think that makes sense right to have it there since it should apply to any evolved op ansatz that if you want the output state to reflect the input one, then the parameters needed to be set are all 0.